no-vary-search.bs

```diff
@@ -189,6 +199,18 @@ The [=obtain a URL search variance=] algorithm ensures that all [=URL search var
   </table>
 </div>
 
+<div algorithm>
+  To <dfn>obtain a URL search variance hint</dfn> given a [=string=] |hintValue|:
+
+  1. Let |fieldValue| be the result of [=parsing structured fields=] given |hintValue| and "`dictionary`".
+  1. If parsing failed, then return null.
+  1. Let |result| be the result of [=parsing a URL search variance=] given |fieldValue|.
+  1. If |result| is null, then return null.
+  1. Otherwise return |result|.
+
+  <p class="note">Speculation rule parsing is strict, so the return value of this algorithm surfaces parsing errors instead of silently returning the [=default URL search variance=].
```

Comment:
Ah, I think I see your point. Also the motivation for the strictness described in "parse speculation rules" talks about misunderstood rules leading to more speculation than was intended. But here, we know the consequence of an invalid hint and it would not cause more speculation.

For debuggability, we should presumably still have the implementation warn about this.

Reply:
Tagging @liviutinta in case of any concerns.

Reply:
I've updated the parsing of the hint to fallback to the default variance. Note that in "parse a speculation rule" this still treats non-string values for expects_no_vary_search as invalid.